src/variableProcessor.ts

```diff
@@ -40,9 +79,130 @@ export class VariableProcessor {
             }
         }
 
+        // promise-based variables using URL-inspection must be last

```

Comment:
I started with `$aadToken` in `getGlobalVariables()`, but it has 2 unique traits compared to other global vars:
1. I need the request URL to tell AAD who the token will be sent to
2. Sign-in uses callbacks

Adding the URL isn't a big deal, but making `getGlobalVariables()` callback-/promise-aware quickly became unnecessarily complicated, so I decided to keep it simple and let this be a unique case. We should revisit this if/when there are other promise-based vars.

Next, I moved this code after global vars are evaluated, which worked great until I added a custom variable in the `aadToken.http` test file (e.g. `GET {{arm}}/subscriptions`). This caused a problem because AAD won't give me a token for `{{arm}}` &ndash; we need to evaluate the custom var _before_ we parse the URL. That's what I meant by "URL-inspection".

Let me know if you think there's a better way to convey that than "URL-inspection". I struggled with it for a while, but moved on hoping I'd come up with a better way to describe it. Clearly, I haven't :-(

Reply:
Oh, understand, totally agree with what you've done here. It makes sense.

Reply:
Changed the comment. Hopefully this helps a little: `// any vars that make decisions on the URL (request var) must be last so all vars have been evaluated`